packages/notte-core/src/notte_core/common/telemetry.py

```
@@ -237,13 +237,15 @@ def decorator(func: F) -> F:
         @wraps(func)
         def wrapper(*args: Any, **kwargs: Any) -> Any:
             event_name = method_name
+            filtered_kwargs = {k: v for k, v in kwargs.items() if k not in exclude_kwargs}
             try:
                 result = func(*args, **kwargs)
-                filtered_kwargs = {k: v for k, v in kwargs.items() if k not in exclude_kwargs}
                 capture_event(event_name, properties={"input": {"args": args, "kwargs": filtered_kwargs}})
                 return result
```

Ruff (0.15.12)

[warning] Consider moving this statement to an `else` block

(TRY300)